content/articles/an-introduction-to-blockchain-architecture/index.md

```diff
@@ -0,0 +1,151 @@
+
+There might be some data in the first(Genesis) block, such as transactions details by cryptocurrency users. It also consists of a Hash, a string of characters derived from the fields in the blockchain but in an encrypted format. The blocks usually contain a previous hash, but the Genesis block doesn’t because it is the first block.
+
+The next block created will have some data in it. The data includes information such as:
+- The hash
+- The previous hash
+
+The previous hash references the hash of the previous block. If a single character in the previous block's data is altered, this will cause the hash to change completely, and the algorithm won’t update it in the current block. Hence, the relationship between them will break.
+
+The above process proceeds as more blocks are mined. Finally, they are linked to one another based on the previous hash. In summary, Blockchain is a series of cryptographically linked blocks with each other through these hashes.
+
+### Components of Blockchain
+The realms of blockchain are simply the characteristics or features that make blockchain what it is today. These characteristics are the founding pillars of blockchain technology, and they are:
+- Immutable Ledger
+- Distributed P2P Network
+- Hash Cryptography
+- Mining and Proof of Work
+- Consensus Protocols
+
+In this article, I will touch briefly on them, and in future articles, I will explain them in detail.
+
+### Immutable Ledger
+
+To understand the concept of the immutable ledger, I’ll add some visuals below:
+
+![Blockchain Visual Representation](/engineering-education/an-introduction-to-blockchain-architecture/blockchain.png)
+
+
+A ledger or block must be immutable. It must not be subject to change. In reality, when we add entries to a ledger, it can be altered, which is a fraudulent act. One of the core pillars of blockchain is to ensure that data in the block is not subject to change, and here is how it guarantee it.
+
+When a block is mined and added to the blockchain, a hash for the block is generated from combining the data in the block.
+
+In the image above, the first block is the genesis block, and it has no previous hash but its current hash. The next block has a previous hash that is the same as the hash of the first block, and it also consists of the current hash. 
+
+The two blocks are chained based on the hashes, and when the data in the first block changes, the hash of the first block changes but not in the second block, which results in the chain between the two blocks breaking.
+
+The blockchain will notify of the change in the block's data due to the hash changing. This feature ensures that the data in the ledger doesn’t change.
+
+### Distributed P2P Network
+
+Here is an image view of what a distributed peer-to-peer network looks like:
```

Comment:
Can you explain how an immutable ledger and distributed p2p network are connected? This needs further clarification